python/egobox/egobox.pyi

```diff
@@ -0,0 +1,488 @@
+from dataclasses import dataclass
+from enum import Enum
+from math import inf
+from typing import TYPE_CHECKING, Callable
+
+if TYPE_CHECKING:
+    from numpy.typing import NDArray
+    import numpy as np
+
+    NDArray1DFloat = NDArray[...]
+    NDArray2DFloat = NDArray[...]
+
+class Sampling(Enum):
+    Lhs = 1
+    FullFactorial = 2
+    Random = 3
+    LhsClassic = 4
+    LhsCentered = 5
+    LhsMaximin = 6
+    LhsCenteredMaximin = 7
+
+def sampling(
+    method: Sampling, xspecs: list[XSpec], n_samples: int, seed: float | None = None
+) -> NDArray[np.float64]:
+    """
+    Generate samples using a specified method.
+
+    Parameters
+    ----------
+    method : {"LHS", "FULL_FACTORIAL", "RANDOM"}
+        The sampling method to use.
+    xspecs : list of XSpec
+        List of XSpec objects describing variable bounds or types.
+    n_samples : int
+        Number of samples to generate.
+    seed : int
+        Random seed for reproducibility.
+
+    Returns
+    -------
+    numpy.ndarray
+        A 2D array of shape (n_samples, n_variables) containing the generated samples.
```

Comment:
This was adapted from what it has in the pyo3 wrapper, thus a bit duplicate. Let me know whether you want to keep it with slightly change the format to be python compatible, or the original one or just removed. I'll change for other function docstrings.

Reply:
for the format, I was refer to this PEP https://peps.python.org/pep-0287/ but I think it is fine to also use markdown one, it doesn't matter too much.

For the duplication, I think I'll remove all docstring from the `pyi` file at the moment, and leave it for the runtime check (for some IDE, it can turn on the runtime check for the docstring) from the pyo3 wrapper. I'll check how to parse from the pyo3 wrapper into a `pyi` stub file if possible. Then it can avoid the duplication. But for me, just the typing in the function signature would helpful enough.